saleor/plugins/webhook/tests/test_shipping_webhook.py

```diff
@@ -1275,6 +1284,29 @@ def test_parse_list_shipping_methods_with_metadata(app):
     assert response[0].description == response_data_with_meta[0]["description"]
 
 
+@mock.patch.object(logger, "warning")
+def test_parse_list_shipping_methods_invalid_currency(mocked_logger, app):
+    # given
+    response_data_with_meta = [
+        {
+            "id": 123,
+            "amount": 10,
+            "currency": "ABC",
+            "name": "shipping",
+            "description": "Description",
+            "maximum_delivery_days": 10,
+            "minimum_delivery_days": 2,
+            "metadata": {"field": "value"},
+        }
+    ]
+    # when
+    response = parse_list_shipping_methods_response(response_data_with_meta, app)
```

Comment:
Opinion:

I'm thinking if `parse_list_shipping_methods_response` shouldn't actually raise? I know it may be only a syntax here if we check against "None", but I think raise+except is more readable that we are processing some invalid state

```
try:
  parse_list_shipping_methods_response(response_data_with_meta, app)

except ValidationError:
  logger.warning(...)
  ...

```

Reply:
I don't understand you concern. 
The `parse_list_shipping_methods_response` is logging the warnings, to not break the previous behaviour.